### DISCUSSION TITLE
Adding devices and entities

### DISCUSSION THREAD

**Discussion by gotta-jibboo** [tsightler/ring-mqtt#495] [Support Q&A]
Hello! Thank you for developing this add-on. I recently installed it with the hopes I'd have access to various sensors and devices within my Ring alarm ecosystem for some status and control within Home Assistant. 

I cannot find new entities and devices, though, even after my installation. Is there a procedure out there on adding entities and devices through this add-on? I cannot find anything already in the discussions.

**Comment by tsightler** [comment]
Are you sure everything is working?  What do the logs say?  Devices should be automatically discovered if things are configured correctly.

**Reply by gotta-jibboo** [reply]
I'm not sure. It looks like in the logs it's recognizing all the different devices within my ring ecosystem. It's reporting various sensors and their battery status. I just don't know how to add them into the devices and entities portion of Home Assistant. 

I had the stock integration of Ring going previously. Would that be doing anything to inhibit HA from recognizing the devices from the new add-on?

**Reply by tsightler** [reply]
Again, you do not need to add anything into devices and entities, if everything is configured correctly then they will show up there automatically.  Since you say the logs appear to show the devices that implies you have the Ring-mqtt and Mosquitto addons working, so I'll guess you've failed to configure the HA Mosquitto integration but it's just a guess.

**Reply by gotta-jibboo** [reply]
I guess I don't know how to configure the Mosquitto. I'm fairly new to HA. I've got Mosquitto broker installed. Is there a tutorial on how to set it up properly to work with the Ring-MQTT add on?

**Reply by gotta-jibboo** [reply]
I found this error in another log. So it looks like maybe my MQTT broker isn't working? 

2022-11-12 15:17:04.498 ERROR (Thread-3 (_thread_main)) [homeassistant.components.mqtt.client] Unable to connect to the MQTT broker: Connection Refused: not authorised.

**Reply by tsightler** [reply]
Just installing the Mosquitto broker addon is not enough, that just provides the broker itself, but then you must also configure the Mosquitto integration in HA, so that HA connects to the broker.  You just need to go to the integration and click configure and walk through the wizard.  The fact that you are seeing error messages from Home Assistant indicates that you haven't yet configured HA to properly connect to the broker.  There's like a million different blogs and videos showing how to configure MQTT, nothing special is needed for ring0mqtt, and actually, based on what you have posted, that side is already working, it's just that HA can't connect to the broker, so this indicates you just need to configure that side.

**Reply by tsightler** [reply]
In general there's not much to this.  Most of the tutorials/videos I've found over-complicate the process and have users do things like setup custom users, etc, all of which is 100% unnecessary and just add complication if you already don't know what you are doing.

If you can't get this to work at all then I would suggest you delete the Mosquitto and ring-mqtt addon, and delete the mosquitto integtation, and restart the process from the start, following exactly that process.  Install Mosquitto addon, wait for the MQTT service to be automatically discovered and appear in the "Setup...Devices...Integrations", click configure and then "Submit" when prompted to HA to connect to the MQTT broker provided by the add-on Mosquitto broker.   Don't change anything, just accept the defaults.  Assuming that is successful then add ring-mqtt, don't change anything, just authenticate with Ring, that's it.

**Reply by gotta-jibboo** [reply]
I got it working! I uninstalled the MQTT stuff and reinstalled it. Now I see them in the MQTT integration. Thanks for your help!

**Reply by tsightler** [reply]
Glad you got it working!
